cdap-ui/app/directives/my-pipeline-configurations/my-batch-pipeline-config/my-batch-pipeline-config.html

```diff
@@ -264,32 +267,44 @@ <h3 class="modeless-title">
-              class="btn btn-secondary"
-              ng-if="BatchPipelineConfigCtrl.activeTab !== 'runtimeArgs' && BatchPipelineConfigCtrl.activeTab !== engineConfig"
-              ng-click="BatchPipelineConfigCtrl.updatePipeline()"
-              ng-disabled="!BatchPipelineConfigCtrl.enablePipelineUpdate || BatchPipelineConfigCtrl.updatingPipeline || BatchPipelineConfigCtrl.startingPipeline"
-            >
-             <span ng-if="!BatchPipelineConfigCtrl.updatingPipeline">Save</span>
-             <span ng-if="BatchPipelineConfigCtrl.updatingPipeline">Saving</span>
-             <span ng-if="BatchPipelineConfigCtrl.updatingPipeline">
-               <span class="fa fa-spinner fa-spin"></span>
-             </span>
-            </button>
+            <div class="apply-run-container">
+              <button
+                class="btn btn-primary apply-run"
+                ng-disabled="BatchPipelineConfigCtrl.buttonsAreDisabled() || BatchPipelineConfigCtrl.startingPipeline || BatchPipelineConfigCtrl.updatingPipeline"
+                ng-click="BatchPipelineConfigCtrl.applyAndRunPipeline()">
+                <span>Save &amp; {{ BatchPipelineConfigCtrl.pipelineAction }}</span>
+                <span ng-if="BatchPipelineConfigCtrl.startingPipeline" class="fa fa-spinner fa-spin"></span>
+              </button>
+              <button
+                class="btn btn-secondary"
+                ng-disabled="BatchPipelineConfigCtrl.buttonsAreDisabled()"
+                ng-if="BatchPipelineConfigCtrl.activeTab === 'runtimeArgs'"
+                ng-click="BatchPipelineConfigCtrl.applyAndClose()">
+                Save
+              </button>
+              <button
+                class="btn btn-secondary"
+                ng-if="BatchPipelineConfigCtrl.activeTab !== 'runtimeArgs' && BatchPipelineConfigCtrl.activeTab !== engineConfig"
+                ng-click="BatchPipelineConfigCtrl.updatePipeline()"
+                ng-disabled="!BatchPipelineConfigCtrl.enablePipelineUpdate || BatchPipelineConfigCtrl.updatingPipeline || BatchPipelineConfigCtrl.startingPipeline"
+              >
+               <span ng-if="!BatchPipelineConfigCtrl.updatingPipeline">Save</span>
+               <span ng-if="BatchPipelineConfigCtrl.updatingPipeline">Saving</span>
+               <span ng-if="BatchPipelineConfigCtrl.updatingPipeline">
+                 <span class="fa fa-spinner fa-spin"></span>
+               </span>
+              </button>
+              <span class="num-runtime-args"
+                    ng-if="BatchPipelineConfigCtrl.activeTab === 'runtimeArgs'">
+                  {{ BatchPipelineConfigCtrl.runtimeArguments.pairs.length }}
+                <span ng-if="BatchPipelineConfigCtrl.runtimeArguments.pairs.length === 1">
+                  runtime argument
+                </span>
+                <span ng-if="BatchPipelineConfigCtrl.runtimeArguments.pairs.length > 1">
+                  runtime arguments
+                </span>
```

Comment:
see if you can use ngPluralize for this

https://docs.angularjs.org/api/ng/directive/ngPluralize